oak-core/src/test/java/org/apache/jackrabbit/oak/namepath/impl/NamePathMapperImplTest.java

```diff
@@ -135,6 +135,26 @@ public void testOakToJcr() {
         }
     }
 
+    @Test
+    public void testOakToExpandedJcr() {
+        assertEquals("/{http://www.example.com/foo}bar", npMapper.getExpandedJcrPath("/oak-foo:bar"));
+        assertEquals("/{http://www.example.com/foo}bar/{http://www.example.com/quu}qux", npMapper.getExpandedJcrPath("/oak-foo:bar/oak-quu:qux"));
+        assertEquals("{http://www.example.com/foo}bar", npMapper.getExpandedJcrPath("oak-foo:bar"));
+        assertEquals(".", npMapper.getExpandedJcrPath(""));
+
+        try {
+            npMapper.getExpandedJcrPath("{http://www.jcp.org/jcr/nt/1.0}unstructured");
```

Comment:
Not sure whether this is good. As a user of this API, I would expect that I can apply it to *any* valid JCR name (in both forms). Otherwise I may have to check before.

Proposal: allow both forms (with one being a NO-OP) and include that case in the Javadoc.

Reply:
Good point, I agree both name forms should be accepted.